### DISCUSSION TITLE
Problem

### DISCUSSION THREAD

**Discussion by flexait-alex** [community/community#123643] [Programming Help]
Good morning!
I'm new here and I'm a little lost, if anyone can help me with the question below, I would be very grateful. 

I'm setting up Ansible on an Ubuntu, I receive the error below after exec my playbook.

ansible$ sudo ansible-playbook ec2_ssm_access.yml
[sudo] password for flexa-aws:
ERROR! couldn't resolve module/action 'amazon.aws.aws_ssm_instance_info'. This often indicates a misspelling, missing collection, or incorrect module path.

The error appears to be in '/etc/ansible/ec2_ssm_access.yml': line 38, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:
    - name: Get EC2 instances information
      ^ here
      
ansible$ ansible-galaxy collection list amazon.aws

# /usr/lib/python3/dist-packages/ansible_collections
Collection Version
---------- -------
amazon.aws 7.5.0

# /usr/local/lib/python3.10/dist-packages/ansible_collections
Collection Version
---------- -------
amazon.aws 7.5.0
/etc/ansible$

**Comment by InWamos** [comment]
Probably, Ansible collection for amazon.aws.aws_ssm_instance_info is not installed or not correctly referenced in your playbook.

```Run the following command: ansible-galaxy collection install amazon.aws```
### You may need to use sudo if necessary.

Also, in your playbook, make sure you're using the fully qualified collection name (FQCN) for the module. For example:
- name: Gather SSM instance info
  amazon.aws.aws_ssm_instance_info:
    filters:
      instance_id:
        - i-1234567890abcdef0

**Reply by flexait-alex** [reply]
Hello InWamos, thanks about your contact.

Below information thats confirm collection installed, 

/etc/ansible$ sudo ansible-galaxy collection install amazon.aws
[sudo] password for flexa-aws:
[DEPRECATION WARNING]: [defaults]collections_paths option, does not fit var naming standard, use the singular form collections_path instead. This
feature will be removed from ansible-core in version 2.19. Deprecation warnings can be disabled by setting deprecation_warnings=False in
ansible.cfg.
Starting galaxy collection install process
Nothing to do. All requested collections are already installed. If you want to reinstall them, consider using `--force`.
/etc/ansible$


flexa-aws@FLEXAIT:/etc/ansible$ ansible-galaxy collection list amazon.aws
[DEPRECATION WARNING]: [defaults]collections_paths option, does not fit var naming standard, use the singular form collections_path instead. This
feature will be removed from ansible-core in version 2.19. Deprecation warnings can be disabled by setting deprecation_warnings=False in
ansible.cfg.

# /usr/lib/python3/dist-packages/ansible_collections
Collection Version
---------- -------
amazon.aws 7.5.0

# /usr/local/lib/python3.10/dist-packages/ansible_collections
Collection Version
---------- -------
amazon.aws 7.5.0
/etc/ansible$

My ansible.cfg:

# for example, for 2.9: https://github.com/ansible/ansible/blob/stable-2.9/examples/ansible.cfg
[defaults]
host_key_checking = False
inventory = hosts.ini
#private_key_file = /path/to/your/private/key
library = /usr/lib/python3/dist-packages/ansible_collections/amazon/aws/plugins/action/
collections_paths = /usr/lib/python3/dist-packages/ansible_collections/

[privilege_escalation]
become = True
become_method = sudo
become_user = root

[aws_vault]
aws_vault_executable = /usr/bin/aws-vault
aws_vault_profile = flexa
plugin_connection = awsssm

[connection]
plugin = amazon.aws.aws_ssm
ansible_connection = amazon.aws.aws_ssm

[ansible_library]

#module_utils = /usr/lib/python3/dist-packages/ansible_collections/amazon/aws/plugins/modules/
module_utils = /home/flexa-aws/.ansible/collections/ansible_collections/community/crypto/plugins/module_utils

information the my playbook:

- hosts: localhost
  connection: local
  gather_facts: false
  collections:
    - amazon.aws

  vars:
    instance_ids:
      - "i-0123456789abcdef"  # Substitua pelos IDs das suas instâncias EC2

  tasks:

    - name: Get EC2 instances information
      amazon.aws.aws_ssm_instance_info:
        filters:
          instance-ids: "{{ instance_ids }}"
      register: ssm_instances

    - name: Connect to EC2 instances via SSM Session Manager
      amazon.aws.aws_ssm_instance_info:
        instance_id: "{{ item.instance_id }}"
        session_name: "Ansible SSM Session"
        working_dir: "/tmp"
        command: "{{ ansible_env.SHELL }}"
      with_items: "{{ ssm_instances.instances }}"
      register: ssm_sessions

    - name: Display SSM Session output
      debug:
        var: item.output
      with_items: "{{ ssm_sessions.results }}"
